src/checker.rs

```diff
@@ -70,9 +114,8 @@ pub fn path_exists(filename: impl AsRef<std::path::Path>) -> bool {
     filename.as_ref().exists()
 }
 
-// TODO: Test
-pub fn get_os_type() -> Result<String> {
-    Ok(sys_info::os_type().map(|os_type| os_type.to_lowercase())?)
+pub fn get_os_type(os_type: String) -> Result<String> {
```

Comment:
Since you don't need ownership of the `String`, it would be better to use `&str` as the parameter type. Every `String` can be dereferenced as a `&str` and the API gets more ergonomic.

Reply:
Also why does this return a `Result`?

Reply:
Thanks, good catch :)
When I adjust the calling function from `let os_type = checker::get_os_type(&sys_info::os_type()?)?;
` to `let os_type = checker::get_os_type(&sys_info::os_type().unwrap());` it's easier to read and handle. This whole String stuff is a bit wierd :)